pkg/querier/queryrange/codec.go

```diff
@@ -173,38 +173,76 @@ func (r *LokiSeriesRequest) LogToSpan(sp opentracing.Span) {
 
 func (*LokiSeriesRequest) GetCachingOptions() (res queryrangebase.CachingOptions) { return }
 
-func (r *LokiLabelNamesRequest) GetEnd() int64 {
-	return r.EndTs.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+// In some other world LabelRequest could implement queryrangebase.Request.
+type LabelRequest struct {
+	path string
+	logproto.LabelRequest
+}
+
+func NewLabelRequest(start, end time.Time, query, name, path string) *LabelRequest {
+	return &LabelRequest{
+		LabelRequest: logproto.LabelRequest{
+			Start:  &start,
+			End:    &end,
+			Query:  query,
+			Name:   name,
+			Values: name != "",
+		},
+		path: path,
+	}
 }
 
-func (r *LokiLabelNamesRequest) GetStart() int64 {
-	return r.StartTs.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+func (r *LabelRequest) AsProto() *logproto.LabelRequest {
+	return &r.LabelRequest
 }
 
-func (r *LokiLabelNamesRequest) WithStartEnd(s int64, e int64) queryrangebase.Request {
-	clone := *r
-	clone.StartTs = time.Unix(0, s*int64(time.Millisecond))
-	clone.EndTs = time.Unix(0, e*int64(time.Millisecond))
-	return &clone
+func (r *LabelRequest) GetEnd() int64 {
+	return r.End.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+}
+
+func (r *LabelRequest) GetEndTs() time.Time {
+	return *r.End
+}
+
+func (r *LabelRequest) GetStart() int64 {
+	return r.Start.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+}
+
+func (r *LabelRequest) GetStartTs() time.Time {
+	return *r.Start
+}
+
+func (r *LabelRequest) GetStep() int64 {
+	return 0
```

Comment:
I find the naming a bit confusing here. Start and End are timestamps, get `GetStart` and `GetEnd` return ints, while you have to call `GetStartTs` and `GetEndTs` to get the same type as the actual property. I feel like `GetStart` and `GetEnd` should return the same type as the actual property, with different names for the methods returning ints, but this is fairly nit-picky.

Reply:
They should but this is part of the `queryrangebase.Request` interface that we should change in another PR.